src/goto-symex/execution_state.cpp

```diff
@@ -239,6 +239,10 @@ void execution_statet::symex_step(reachability_treet &art)
   // case or forward condition
   if((base_case || forward_condition) && instruction.inductive_step_instruction)
   {
+    // This assertion will prevent us of having weird side-effects (issue #538)
```

Comment:
What do you mean by "weird side-effects"?

Reply:
Skipping instructions during incremental. https://github.com/esbmc/esbmc/issues/538

Reply:
Can you say that in the comment?